internal/resources/google/compute_cost_component_helpers.go

```diff
@@ -64,6 +66,85 @@ func computeCostComponent(region, machineType string, purchaseOption string, ins
 	}
 }
 
+// customComputeCostComponent returns a cost component for custom Compute instance usage.
+func customComputeCostComponents(region, machineType string, purchaseOption string, instanceCount int64, monthlyHours *float64) []*schema.CostComponent {
+	sustainedUseDiscount := 0.0
+	fixPurchaseOption := "preemptible"
+	if strings.ToLower(purchaseOption) == "on_demand" {
+		fixPurchaseOption = "OnDemand"
+		switch strings.ToLower(strings.Split(machineType, "-")[0]) {
+		case "c2", "n2", "n2d":
+			sustainedUseDiscount = 0.2
+		case "n1", "f1", "g1", "m1":
+			sustainedUseDiscount = 0.3
+		}
+	}
+
+	var re = regexp.MustCompile(`(\D.+)-(\d+)-(\d.+)`)
+	strCPUAmount := re.ReplaceAllString(machineType, "$2")
+	strRAMAmount := re.ReplaceAllString(machineType, "$3")
```

Comment:
I am struggling to come up with a regex that will validate and parse different possibilities like:
```
custom-20-20480
custom-20-20480-ext
n2-custom-20-20480
n2-custom-20-20480-ext
n2d-custom-20-20480
n2d-custom-20-20480-ext
e2-custom-20-20480
e2-custom-20-20480-ext
```

Reply:
Could something like this work?

```go
var strCPUAmount, strRAMAmount string

re := regexp.MustCompile(`custom-(\d+)-(\d+)`)
matches := re.FindStringSubmatch(machineType)

if len(matches) > 2 {
    strCPUAmount = matches[1]
    strRAMAmount = matches[2]
}
```

Reply:
I don't think so because we'll need to either extract or not the instance type.. when beginning with `custom`, we assume the type as `N1`, otherwise, we need to get it from the first position before `-` in order to enrich the GraphQL query and bring more accurate results.

Reply:
Not the most elegant solution but it worked for the cases I foresaw about custom instances: https://github.com/infracost/infracost/pull/2004/commits/8f8ebf3cc1afc1bdd7208aef044b5f2f396550a2

Having this input:
```hcl
resource "google_container_node_pool" "dev-1" {
  cluster  = "dev"
  name     = "dev-1"
  location = "europe-central2-c"

  autoscaling {
    max_node_count = 20
    min_node_count = 1
  }

  management {
    auto_repair  = true
    auto_upgrade = true
  }

  upgrade_settings {
    max_surge       = 1
    max_unavailable = 0
  }

  node_config {
    preemptible = false
    oauth_scopes = [
      "https://www.googleapis.com/auth/cloud-platform"
    ]

    machine_type = "custom-20-20480"

    image_type   = "COS_CONTAINERD"
    disk_type    = "pd-standard"
    disk_size_gb = "100"

    labels = {
      "pool" = "dev-1"
    }

    workload_metadata_config {
      mode = "GKE_METADATA"
    }
  }
}

resource "google_container_node_pool" "dev-2" {
  cluster  = "dev"
  name     = "dev-2"
  location = "europe-central2-c"

  autoscaling {
    max_node_count = 20
    min_node_count = 1
  }

  management {
    auto_repair  = true
    auto_upgrade = true
  }

  upgrade_settings {
    max_surge       = 1
    max_unavailable = 0
  }

  node_config {
    preemptible = false
    oauth_scopes = [
      "https://www.googleapis.com/auth/cloud-platform"
    ]

    machine_type = "custom-20-20480-ext"

    image_type   = "COS_CONTAINERD"
    disk_type    = "pd-standard"
    disk_size_gb = "100"

    labels = {
      "pool" = "dev-2"
    }

    workload_metadata_config {
      mode = "GKE_METADATA"
    }
  }
}


resource "google_container_node_pool" "dev-3" {
  cluster  = "dev"
  name     = "dev-3"
  location = "europe-central2-c"

  autoscaling {
    max_node_count = 20
    min_node_count = 1
  }

  management {
    auto_repair  = true
    auto_upgrade = true
  }

  upgrade_settings {
    max_surge       = 1
    max_unavailable = 0
  }

  node_config {
    preemptible = false
    oauth_scopes = [
      "https://www.googleapis.com/auth/cloud-platform"
    ]

    machine_type = "n2-custom-20-20480"

    image_type   = "COS_CONTAINERD"
    disk_type    = "pd-standard"
    disk_size_gb = "100"

    labels = {
      "pool" = "dev-3"
    }

    workload_metadata_config {
      mode = "GKE_METADATA"
    }
  }
}

resource "google_container_node_pool" "dev-4" {
  cluster  = "dev"
  name     = "dev-4"
  location = "europe-central2-c"

  autoscaling {
    max_node_count = 20
    min_node_count = 1
  }

  management {
    auto_repair  = true
    auto_upgrade = true
  }

  upgrade_settings {
    max_surge       = 1
    max_unavailable = 0
  }

  node_config {
    preemptible = false
    oauth_scopes = [
      "https://www.googleapis.com/auth/cloud-platform"
    ]

    machine_type = "n2-custom-20-20480-ext"

    image_type   = "COS_CONTAINERD"
    disk_type    = "pd-standard"
    disk_size_gb = "100"

    labels = {
      "pool" = "dev-4"
    }

    workload_metadata_config {
      mode = "GKE_METADATA"
    }
  }
}
```

I managed to get these results. But I am not 100% confident about them as I saw in the cloud pricing database Extended CPU without an actual price. Because of that, I left the Extended option only for RAM. 
```bash
$ make run ARGS="breakdown --path ~/l/infracost-gcp-custom-instances/terraform"
env INFRACOST_ENV=dev go run -ldflags="-X 'github.com/infracost/infracost/internal/version.Version='" github.com/infracost/infracost/cmd/infracost breakdown --path ~/l/infracost-gcp-custom-instances/terraform
...
Evaluating Terraform directory at /home/tom/l/infracost-gcp-custom-instances/terraform
  ✔ Downloading Terraform modules
  ✔ Evaluating Terraform directory
  ✔ Retrieving cloud prices to calculate costs

Project: /home/tom/l/infracost-gcp-custom-instances/terraform

 Name                                                                          Monthly Qty  Unit           Monthly Cost

 google_container_node_pool.dev-1
 ├─ Standard provisioned storage (pd-standard)                                         300  GB                   $14.40
 ├─ Custom Instance CPU usage (Linux/UNIX, on-demand, custom-20-20480)               2,190  hours             $1,751.69
 └─ Custom Instance RAM usage (Linux/UNIX, on-demand, custom-20-20480)               2,190  gibibyte hour       $238.84

 google_container_node_pool.dev-2
 ├─ Standard provisioned storage (pd-standard)                                         300  GB                   $14.40
 ├─ Custom Instance CPU usage (Linux/UNIX, on-demand, custom-20-20480-ext)           2,190  hours             $1,751.69
 └─ Custom Instance RAM usage (Linux/UNIX, on-demand, custom-20-20480-ext)           2,190  gibibyte hour       $512.90

 google_container_node_pool.dev-3
 ├─ Standard provisioned storage (pd-standard)                                         300  GB                   $14.40
 ├─ Custom Instance CPU usage (Linux/UNIX, on-demand, n2-custom-20-20480)            2,190  hours             $1,303.73
 └─ Custom Instance RAM usage (Linux/UNIX, on-demand, n2-custom-20-20480)            2,190  gibibyte hour       $191.07

 google_container_node_pool.dev-4
 ├─ Standard provisioned storage (pd-standard)                                         300  GB                   $14.40
 ├─ Custom Instance CPU usage (Linux/UNIX, on-demand, n2-custom-20-20480-ext)        2,190  hours             $1,303.73
 └─ Custom Instance RAM usage (Linux/UNIX, on-demand, n2-custom-20-20480-ext)        2,190  gibibyte hour       $356.99

 OVERALL TOTAL                                                                                                $7,468.25
──────────────────────────────────
4 cloud resources were detected:
∙ 4 were estimated, all of which include usage-based costs, see https://infracost.io/usage-file
```

cc @aliscott